controllers/azuremachine_controller.go

```diff
@@ -217,20 +217,12 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
 		return reconcile.Result{}, err
 	}
 
-	// TODO(ncdc): move this validation logic into a validating webhook
-	if errs := r.validateUpdate(&machineScope.AzureMachine.Spec, vm); len(errs) > 0 {
-		agg := kerrors.NewAggregate(errs)
-		r.Recorder.Eventf(machineScope.AzureMachine, corev1.EventTypeWarning, "InvalidUpdate", "Invalid update: %s", agg.Error())
-		return reconcile.Result{}, nil
-	}
-
 	// Make sure Spec.ProviderID is always set.
 	machineScope.SetProviderID(fmt.Sprintf("azure:////%s", vm.ID))
 
 	// Proceed to reconcile the AzureMachine state.
 	machineScope.SetVMState(vm.State)
 
-	// TODO(vincepri): Remove this annotation when clusterctl is no longer relevant.
```

Comment:
@vincepri is this still relevant?

Reply:
I don't think we use this any longer, cc @fabriziopandini

Reply:
clusterctl uses `cluster.x-k8s.io/provider` label, so AFAIK this annotation can be removed

Reply:
hmm I removed it and now E2E is failing so maybe it is needed...